lib/python/rose/bush.py

```diff
@@ -518,6 +519,20 @@ def view(self, user, suite, path, path_in_tar=None, mode=None):
         else:
             file_content = self.suite_engine_proc.is_conf(path)
         template = self.template_env.get_template("view.html")
+
+        # convert url(s) to hyperlinks if in text mode
+        if mode == None:
+            urlregex = re.compile('\\b(?:https?|ftp)://\S+\\b')
+            for i in range(0, len(lines)):
+                line = lines[i]
+                match = urlregex.search(line)
+                if match:
+                    begining, end = match.span()
+                    lines[i] = '%s'*3 % (
+                        line[:begining],
+                        '<a href="{0}">{0}</a>'.format(match.group(0)),
+                        line[end:])
```

Comment:
I like the intention of this change, but I think this is leaking template logic into this module. Jinja2 appears to have a `urlize` filter that should allow us to do this sort of stuff within the template.